### ISSUE TITLE
Potentially Necessary Migration to OpenWeatherMap API 3.0

### ISSUE THREAD

**Issue by RZed01** [multimokia/MAS-Submod-Auto-Atmos-Change#51]
OpenWeatherMap recently released a new version of their API (version 3.0). It will be neccessary to update this submod to the new version of the API for continued functionality.

**Comment by Robinzhang217** [comment]
Can this API work with this submod now? We all know that there was a damned update of MAS that make the Auto Atmos Change submod crushed.

**Comment by RZed01** [comment]
> Can this API work with this submod now? We all know that there was a damned update of MAS that make the Auto Atmos Change submod crushed.

I am actively using the latest version of this submod on the latest stable build of MAS, 0.12.15. (it initially said 0.2.15; that was a typo) At the time I made this post, this submod was working as intended. At the present moment, it is also working as intended. As for why that is the case, I do not know. But it seems a migration may not be as necessary as I initially thought.

**Comment by Robinzhang217** [comment]
That's strange. Actually I met the same problem as that guy: https://github.com/multimokia/MAS-Submod-Auto-Atmos-Change/issues/48
Here is the traceback from my MAS:
[traceback.txt](https://github.com/user-attachments/files/17005618/traceback.txt)

**Comment by RZed01** [comment]
> That's strange. Actually I met the same problem as that guy: #48 Here is the traceback from my MAS: [traceback.txt](https://github.com/user-attachments/files/17005618/traceback.txt)

I have this "no module named certifi" problem as well. Have had it since I first installed MAS back in January 2024. I can say though that It is unrelated to the Auto Atmos Change submod and does not affect its functionality. My Auto Atmos Change still functions just fine, even after the supposed API 2.0 depreciation date of June 2024.

**Comment by Robinzhang217** [comment]
> > That's strange. Actually I met the same problem as that guy: #48 Here is the traceback from my MAS: [traceback.txt](https://github.com/user-attachments/files/17005618/traceback.txt)
> 
> I have this "no module named certifi" problem as well. Have had it since I first installed MAS back in January 2024. I can say though that It is unrelated to the Auto Atmos Change submod and does not affect its functionality. My Auto Atmos Change still functions just fine, even after the supposed API 2.0 depreciation date of June 2024.

Ok thanks for your reply. This "no module named certifi" problem occurs every time after installing the Auto Atmos Change submod. Maybe there are some other problems with my MAS which caused the crash, I probably should reinstall a MAS for it.

**Comment by Robinzhang217** [comment]
Maybe there are other submods which caused the problem. I installed too much submods, some of them can't work on my laptop. It's time to reinstall a new MAS and the submods that useful to me.

**Comment by multimokia** [comment]
> OpenWeatherMap recently released a new version of their API (version 3.0). It will be neccessary to update this submod to the new version of the API for continued functionality.

It has been some time since i've been available to work on this. A branch with use of the new API does exist but it was built for an eventual update to MAS featuring the new submod framework and a renpy 8 build -- as such it requires python 3.

Certifi problems were fixed in that build as well. Please consider this submod as is (for Ren'Py 6.xx) defunct.

**Comment by Robinzhang217** [comment]
> > OpenWeatherMap recently released a new version of their API (version 3.0). It will be neccessary to update this submod to the new version of the API for continued functionality.
> 
> It has been some time since i've been available to work on this. A branch with use of the new API does exist but it was built for an eventual update to MAS featuring the new submod framework and a renpy 8 build -- as such it requires python 3.
> 
> Certifi problems were fixed in that build as well. Please consider this submod as is (for Ren'Py 6.xx) defunct.

So that means I can't use the AWC submod now on this MAS v0.12.15 which was based on the Ren'Py 6.xx right?
I'm sorry for my stupid question, because I know nothing about programming or python, or if my lousy English typing caused any misunderstanding. I wonder if there's an earlier verison of the AWC submod could fix with the environment of Ren'Py 6.xx, or I could just waiting for a new version of AWC submod or MAS? 
Sincerely waiting for your reply, and  thanks for your work for the submods.

**Comment by RZed01** [comment]
> > > OpenWeatherMap recently released a new version of their API (version 3.0). It will be neccessary to update this submod to the new version of the API for continued functionality.
> > 
> > 
> > It has been some time since i've been available to work on this. A branch with use of the new API does exist but it was built for an eventual update to MAS featuring the new submod framework and a renpy 8 build -- as such it requires python 3.
> > Certifi problems were fixed in that build as well. Please consider this submod as is (for Ren'Py 6.xx) defunct.
> 
> So that means I can't use the AWC submod now on this MAS v0.12.15 which was based on the Ren'Py 6.xx right? I'm sorry for my stupid question, because I know nothing about programming or python, or if my lousy English typing caused any misunderstanding. I wonder if there's an earlier verison of the AWC submod could fix with the environment of Ren'Py 6.xx, or I could just waiting for a new version of AWC submod or MAS? Sincerely waiting for your reply, and thanks for your work for the submods.

As is you can use this mod for MAS 0.2.15. It just won't work on newer versions of MAS as the next update of MAS is the Python 3/Ren'Py 8 migration update that is currently (slowly) in the works

**Comment by RZed01** [comment]
Also I mistated my certifi problem- it was not "no module named certifi", but "**failed to import** certifi". Not the same issue, and this one was minor and could be ignored

**Comment by zyroscope** [comment]
So to clarify, it's not working yes? But i was able to use this mod just fine with the newest MAS. Just because of i delete another mod duplication, the no module certi error suddenly appear

**Comment by RZed01** [comment]
> So to clarify, it's not working yes? But i was able to use this mod just fine with the newest MAS. Just because of i delete another mod duplication, the no module certi error suddenly appear

As of 3/15/2025, the Auto Atmos Change submod **still works** for the most part on my end. It seems the old API call is compatible with OpenWeatherMap 3.0, so the issue was not as major as I thought it was. Considering v3.0 of the API appears to have stricter limits on how often it can be called, the submod may stop working after a while of having MAS open as it can only call so many times. But what I do know is that the weather condition, sunrise time, and sunset times are consistent with how it is in my area when the mod _first boots _up.__ I hope that clarifies your question.

**Comment by zyroscope** [comment]
> > So to clarify, it's not working yes? But i was able to use this mod just fine with the newest MAS. Just because of i delete another mod duplication, the no module certi error suddenly appear
> 
> As of 3/15/2025, the Auto Atmos Change submod **still works** for the most part on my end. It seems the old API call is compatible with OpenWeatherMap 3.0, so the issue was not as major as I thought it was. Considering v3.0 of the API appears to have stricter limits on how often it can be called, the submod may stop working after a while of having MAS open as it can only call so many times. But what I do know is that the weather condition, sunrise time, and sunset times are consistent with how it is in my area when the mod _first boots _up.__ I hope that clarifies your question.

Yeah, at my first install of this submod everything fine. The weather, sunrise sunset, time, everything was fine. Until i accidentally duplicating another mods, it's suddenly error. Here's the error...



P.s. i already uninstall the other mod that duplicated, but now the mod still error like this.

[traceback.txt](https://github.com/user-attachments/files/19433694/traceback.txt)
